### PR TITLE
When sending a signal to container, only set container to exit on next if the signal is KILL or INT

### PR DESCRIPTION
**Pull request by ble**

### CODE REVIEW

**Comment by ble** [comment]
One way to address #11065 

Not necessarily the best.

**Comment by cpuguy83** [comment]
Is there an open issue that this is fixing?
Can you provide a test in integration-cli?

File: daemon/container.go [hunk 1]
```diff
@@ -715,7 +715,9 @@ func (container *Container) KillSig(sig int) error {
 
 	// signal to the monitor that it should not restart the container
 	// after we send the kill signal
-	container.monitor.ExitOnNext()
+	if sig == syscall.SIGKILL || sig == syscall.SIGINT {
```
Comment (cpuguy83) [inline daemon/container.go]: How about sigterm as well?

**Comment by jfrazelle** [comment]
wait wait dont close

**Comment by jfrazelle** [comment]
just focrce push your branch

**Comment by ble** [comment]
it won't let me reopen because I force-pushed. "dog-wearing-goggles.jpeg"

**Comment by jfrazelle** [comment]
you can open a new one, just dont close it hahaha

**Pull request closed without merging**